### DISCUSSION TITLE
How to diagnose system crashes

### DISCUSSION THREAD

**Discussion by anultravioletaurora** [harvester/harvester#7033] [Q&A] [answered]
I have a bare metal install that is crashing randomly. I've gone through the debug boot setting to try and output a log to /var/crash/ but nothing is output there when the system starts back up again. This is on Harvester 1.3.2

I haven't ruled out a hardware issue, but I'm looking for some help on troubleshooting 🙏 Are there any other logs that I can look at to see what's going on?

**Comment by ibrokethecloud** [comment]
are you please able to confirm if this was a hardware issue?

**Reply by anultravioletaurora** [reply]
Hi there! 
It doesn't seem to be :( I've tried varying combinations of hardware at this point, I'm convinced there's a software needle in this haystack

MemTest86 reports back okay, would there be a log indicating if the network card is failing? That's the only thing I haven't been able to confirm is working

**Comment by anultravioletaurora** [comment]
Hi there! I've done a lot of troubleshooting but I haven't figured it out yet. I've reset my BIOS, moved to a new disk, and got a new PSU, but my node is still randomly rebooting. I'm also having issues enabling the Monitoring addon as well. The graphs just say "Failed to load"

EDIT: Updated with some comments from the support bundle, it doesn't look like the node is fully rebooting, just that the Rancher System Agent is exiting

**Comment by anultravioletaurora** [comment]
I have a support bundle, but it's larger than the Github file attachment limit. I can email it to you all again if you like, @ibrokethecloud :)

**Comment by anultravioletaurora** [comment]
Looking in the support bundle I generated, I can see that it looks like the node is having periodic connection issues to the Harvester Kubernetes cluster?:

`Nov 30 15:18:39 ganymede rancher-system-agent[2133]: time="2024-11-30T15:18:39Z" level=fatal msg="error while connecting to Kubernetes cluster: Get \"https://10.53.235.196/version\": dial tcp 10.53.235.196:443: connect: connection timed out"

Nov 30 15:18:39 ganymede systemd[1]: rancher-system-agent.service: Main process exited, code=exited, status=1/FAILURE

Nov 30 15:18:39 ganymede systemd[1]: rancher-system-agent.service: Failed with result 'exit-code'.

Nov 30 15:18:44 ganymede systemd[1]: rancher-system-agent.service: Scheduled restart job, restart counter is at 1.

Nov 30 15:18:44 ganymede systemd[1]: Stopped Rancher System Agent.
`

**Reply by ibrokethecloud** [reply]
that should not cause a node to reboot.

**Reply by ibrokethecloud** [reply]
are you able to please email this to [harvester-support-bundle@suse.com](mailto:harvester-support-bundle@suse.com)

**Reply by anultravioletaurora** [reply]
Here's a [Google Drive link](https://drive.google.com/file/d/11s2eNPiv0w8m3lyzw4PMY3HmNJubbLEL/view?usp=drive_link) to the bundle!

**Reply by ibrokethecloud** [reply]
there is nothing obvious in the logs but I do see gpu passthrough enabled on ganymede

```
europa-000010000     0000:10:00.0   10de        2882        europa      VGA compatible controller: NVIDIA Corporation AD107 [GeForce RTX 4060]                                                       vfio-pci
europa-000010001     0000:10:00.1   10de        22be        europa      Audio device: NVIDIA Corporation Device 22be                                                                                 vfio-pci
ganymede-000001000   0000:01:00.0   10de        2882        ganymede    VGA compatible controller: NVIDIA Corporation AD107 [GeForce RTX 4060]                                                       vfio-pci
ganymede-000001001   0000:01:00.1   10de        22be        ganymede    Audio device: NVIDIA Corporation Device 22be                                                                                 snd_hda_intel
```

are you please able to please perform the following steps

* shutdown VM with device passthrough on ganymede
* disable the device passthrough for devices: ganymede-000001000  and ganymede-000001001
* delete the pcidevices `kubectl delete pcidevice ganymede-000001000 ganymede-000001001`
* reboot ganymede, and wait for pcidevice objects to be recreated
* re-enable pcidevice passthrough for gpu and audio device and attach them to the VM

We have seen in the past in certain scenarios when a pcidevice is unbound from vfio driver but is in use it can cause the kernel to panic. For some reason the pcidevice cotnroller is trying to bind the audio device to `snd_hda_intel` driver while it is being passed through so could be causing this issue.

**Reply by anultravioletaurora** [reply]
I gave that a shot but I'm stuck now on the same issue that I was having in #6995 where the VGA controller just stays at "In Progress"

![image](https://github.com/user-attachments/assets/17311338-74fc-4c9d-8326-5585aee1902d)

I tried doing my finalizer deletion trick, but when that got it deleted and I tried to re-enable it it didn't take :\

https://drive.google.com/file/d/1LgR5Y2a0ErLvfxxrP0cQYzVhX9qm-3Nc/view?usp=drive_link

**Reply by anultravioletaurora** [reply]
AHHHH so I was trying to enable it as a SAML auth'd user in Rancher. Once I switched over to the harvester VIP and enabled the device then it was passed through successfully :+1:

**Reply by anultravioletaurora** [reply]
I seem to still be having some intermittent health issues with my GPU enabled nodes:

https://drive.google.com/file/d/1LgR5Y2a0ErLvfxxrP0cQYzVhX9qm-3Nc/view?usp=drive_link
https://drive.google.com/file/d/1Xe5xbhjw6gWOXjuGKLw0IwmCTTpbImWV/view?usp=drive_link

**Reply by anultravioletaurora** [reply]
I caught both of those right after the nodes came back to good health

**Comment by anultravioletaurora** [comment]
I can see in the logs that fluentd is being killed?
`ganymede kernel: oom_reaper: reaped process 17478 (fluent-bit)`
Would that have anything to do with my node rebooting at all?

**Comment by anultravioletaurora** [comment, marked as answer]
Found the culprit! Swapping out the NIC with a new one got my node running stable. Thank you all for your help! @ibrokethecloud
